drivers/timer/nrf_rtc_timer.c

```diff
@@ -16,6 +16,7 @@
 #define RTC NRF_RTC1
 
 #define COUNTER_MAX 0x00ffffff
+#define COUNTER_HALF_SPAN BIT(23)
```

Comment:
`COUNTER_MAX/2` would be better

Reply:
I don't really agree.  The counter span is 2^24 bits.  The maximum representable value is 2^24-1, but half the span is 2^23.  If the goal is to detect when we're half-way to a wrap the half-span is the right value to use.

I'd have made `COUNTER_MAX` `(BIT(24)-1U)` but didn't want to change that much.  Without that change `COUNTER_MAX/2` is perhaps a better choice.

It's not worth arguing about.  Any deadline that's far enough in advance of a wrap that the system can spit out the unannounced ticks before they get lost is sufficient.

Pick version 1:
```
#define COUNTER_MAX 0x00ffffff
#define COUNTER_ELAPSE_MAX (COUNTER_MAX / 2)
```
or version 2:
```
#define COUNTER_SPAN BIT(24)
#define COUNTER_MAX (COUNTER_SPAN - 1U)
#define COUNTER_HALF_SPAN (COUNTER_SPAN / 2U)
```
or propose an alternative,

Reply:
what i meant was that there should be relation between `COUNTER_HALF_SPAN` and `COUNTER_MAX`. any version is ok.

Reply:
done (went with version 2).